### PR TITLE
Save ranking in local storage

### LINKED ISSUE #11: Logic | Save rank info in Local Storage
old hash: #24
Acceptance Criteria:
Verify or consult before submitting the task!

- [x] A function that allows you to save a player's nickname and his score in the browser's LocalStorage.
- [x] The function allows you to read the ranking for a given game mode from local storage.
- [x] The ranking should not be global but separate for each game mode.
- [x] Component correctness covered by unit tests.

### PR DESCRIPTION
**Pull request by synowa**
closes #11 fixes #11

### CODE REVIEW

**Review by Enessetere — APPROVED** [review]

File: src/app/RankStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const maxRankingSize = 3;
+
+export const getRanking = (mode) =>
+  JSON.parse(localStorage.getItem(mode)) || [];
+
+export const rankPlayer = (nickname, score, mode) => {
+  const currentRanking = getRanking(mode);
+  const newRanking = [...currentRanking, { nickname, score }]
```
Comment (Rednaxela700) [inline src/app/RankStore.js:8]: How about
```js
if (quizRanking.length > 3) {
//shorten array to 3 items
} 
```

File: src/app/RankStore.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const maxRankingSize = 3;
+
+export const getRanking = (mode) =>
+  JSON.parse(localStorage.getItem(mode)) || [];
+
+export const rankPlayer = (nickname, score, mode) => {
```
Comment (Rednaxela700) [inline src/app/RankStore.js:6]: You could use destructuring in parameter:
`export const rankPlayer = ({nickname, score, mode})`
or take that info from a summary object
`export const rankPlayer = (gameSummary)`

File: src/app/RankStore.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const maxRankingSize = 3;
```
Comment (Rednaxela700) [inline src/app/RankStore.js:1]: This value is used only once, are you going to use it somewhere else?

**Review by Rednaxela700 — COMMENTED** [review]
Pretty good. I hope we could engage in further conversation about this. Btw, I think that the test files deserves separate commits.

**Review by DominikNowak — APPROVED** [review]
Looks good. I am impressed with the unit tests. Good job :)

File: src/app/RankStore.js (hunk 2 above)
Comment (synowa) [reply src/app/RankStore.js:6]: Do you want me to do that or it's just a suggestion?

File: src/app/RankStore.js (hunk 3 above)
Comment (synowa) [reply src/app/RankStore.js:1]: No, it's not used anywhere else. Should I remove it?
I thought of this as a good practice to make that kind of variables easy to change.

File: src/app/RankStore.js (hunk 1 above)
Comment (synowa) [reply src/app/RankStore.js:8]: `export const rankPlayer = (nickname, score, mode) => {
  const currentRanking = getRanking(mode);
  let newRanking = [...currentRanking, { nickname, score }].sort(
    (a, b) => b.score - a.score,
  );
  if (newRanking.length > 3) {
    newRanking = newRanking.slice(0, maxRankingSize);
  }
  localStorage.setItem(mode, JSON.stringify(newRanking));
};`

Do you mean something like this?

**Pull request merged**